### PR TITLE
Clear search results after navigating

### LINKED ISSUE #24466: Clicking on Explore while in search should clear search query
### Steps to reproduce the problem

1. Search for something like 'test'
2. Click Explore in the side navigation column

### Expected behaviour

To see Posts / Hashtags / News / For you that are trending

### Actual behaviour

Seeing All / Profiles / Hashtags / Posts for the search results

### Detailed description

_No response_

### Specifications

Mastodon 4.1.2
Bird ui, but also occurs on Mastodon social etc

### PR DESCRIPTION
**Pull request by TheDevJoao**
This PR fixes [#24466](https://github.com/mastodon/mastodon/issues/24466)

Previously, if something was searched `Search results` would be prioritized over the explorer page:

production_search

The only way to render the explorer page after executing a search query would be by clicking the icon with the id: `times-circle` in the search bar as shown above.

This PR calls the clearSearch method from the `mastodon/actions` directory in the `NavigationPanel` component via `onClick`  to clear the search query and allow the user to access the explorer page.

Here's a preview of the fixed behavior:
fixed_search2

### CODE REVIEW

**Comment by TheDevJoao** [comment]
@ClearlyClaire could you review this whenever you have time? 😃

File: app/javascript/mastodon/features/explore/index.jsx [hunk 1]
```diff
@@ -45,6 +51,10 @@ class Explore extends React.PureComponent {
     this.column = c;
   };
 
+  componentWillUnmount () {
```
Comment (nschonni) [inline app/javascript/mastodon/features/explore/index.jsx:54]: I think this lifecycle script is deprecated. Currently tracking the removal in https://github.com/mastodon/mastodon/pull/24471

Comment (TheDevJoao) [reply app/javascript/mastodon/features/explore/index.jsx:54]: Thanks for the quick response, @nschonni! 

Hmm, I'm pretty limited when it comes to React class components 😓, any idea on which method would substitute `componentWillUnmount` in this case?

**Comment by Gargron** [comment]
What if I click on a search result and then want to navigate back?

**Comment by TheDevJoao** [comment]
Hey @Gargron! Navigating back to the search results is still the same:

search_navigate_back

**Comment by TheDevJoao** [comment]
@Gargron any chance you can review this PR?

File: app/javascript/mastodon/features/explore/index.jsx
Comment (ClearlyClaire) [inline app/javascript/mastodon/features/explore/index.jsx:1]: The `onLoadClear` property introduced in this file isn't used anywhere, I think the changes made in this file should be discarded.

File: app/javascript/mastodon/features/ui/components/__tests__/clear-search.jsx
Comment (ClearlyClaire) [inline app/javascript/mastodon/features/ui/components/__tests__/clear-search.jsx:1]: I do not understand what this test is meant to test exactly.

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]

**Comment by mjankowski** [comment]
Closing due to lack of activity, CI/lint/annotation issues, and unresponded to review feedback.

That said - this seems like a good idea - if you want to move forward, can you rebase, re-open, address feedback, etc?

**Pull request closed without merging**